Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalEnumerate.mm

```diff
@@ -125,3 +125,9 @@
         records.emplace_back(devRec);
     }
 }
+
+MTL::Device* plMetalEnumerate::DeviceForDisplay(hsDisplayHndl display)
+{
+    id<MTLDevice> device = CGDirectDisplayCopyCurrentMetalDevice(display);
+    return static_cast<MTL::Device*>(device);
+}
```

Comment:
This is required because there isn't a good way to pass the Metal device from the enumeration types to the pipeline. The display id gets passed, so this is a convenience function to get the Metal device from the display passed to the pipeline.

I'm aware this is muddying the waters a bit with the Obj-C use in this file (this is case back to MTL::Device for the pipeline.) In since the file still has to be Obj-C++ to support CGDirectDisplayCopyCurrentMetalDevice without starting to hack things I'm not sure yet what I'll do here.